pkg/daemon/daemon.go

```diff
@@ -1406,6 +1406,9 @@ func checkV3Units(units []ign3types.Unit) error {
 	for _, u := range units {
 		for j := range u.Dropins {
 			path := filepath.Join(pathSystemd, u.Name+".d", u.Dropins[j].Name)
+			if u.Dropins[j].Contents == nil || *u.Dropins[j].Contents == "" {
```

Comment:
Doesn't this technically allow the error in the bug to slip through in the future? i.e. if I deleted a dropin (or made it empty), we never verify if the file itself was properly deleted?

Reply:
Moving this over the `deleteStaleData` should address that.